backend/handlers/markets/listmarkets.go

```diff
@@ -52,7 +52,11 @@ func ListMarketsHandler(w http.ResponseWriter, r *http.Request) {
 		marketVolume := marketmath.GetMarketVolume(bets)
 		lastProbability := probabilityChanges[len(probabilityChanges)-1].Probability
 
-		creatorInfo := usersHandlers.GetPublicUserInfo(db, market.CreatorUsername)
+		creatorInfo, err := usersHandlers.GetPublicUserInfo(db, market.CreatorUsername)
```

Comment:
I think this might be a common issue if the creator decides to delete their account. We could create a null user to handle this case or just prevent account deletion unless the user releases the market manually whether by deleting the market or changing "ownership"

Reply:
1. 404 Error Handling: You are correct that it should be a 404 error when a user is not found, and I will update the logic to reflect that.

2. Immutable Usernames for Traceability: Usernames should be immutable to maintain traceability of market activities. Allowing users to delete their accounts would remove critical records of the markets they created and how they resolved. Instead, we can implement an archiving system that retains the username but removes sensitive personal information like the display name. This would help prevent users from creating unreliable markets and then disappearing without accountability, ensuring the integrity of the platform.

Put together a decision ticket about this here: https://github.com/orgs/openpredictionmarkets/projects/9/views/1?pane=issue&itemId=80893304